core/src/main/scala/akka/kafka/ConnectionCheckerSettings.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 - 2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.kafka
+
+import akka.util.JavaDurationConverters._
+import com.typesafe.config.Config
+
+import scala.concurrent.duration.FiniteDuration
+
+sealed abstract class ConnectionCheckerSettings private[kafka] (val enable: Boolean)
+
+object ConnectionCheckerSettings {
+
+  val configPath: String = "connection-checker"
+  val fullConfigPath: String = ConsumerSettings.configPath + "." + configPath
+
+  /**
+   * Create settings from a configuration with layout `connection-checker`.
+   */
+  def apply(config: Config): ConnectionCheckerSettings = {
+    val enable = if (config.hasPath("enable")) config.getBoolean("enable") else false
+    if (enable) EnabledConnectionCheckerSettings(config) else DisabledConnectionCheckerSettings
```

Comment:
Thank you for trying this pattern out. I believe it would be just enough to have one implementation with an `enabled` field to decide if the actor should spin up. This API looks a bit too large for the use-case.

Reply:
I don't understand what you exactly mean. I can't create one settings class because in one case I expect configs and `enabled: true`, in another configs can not be present at all. So I can't create Settings like 
```
ConnectionCheckerSettings(
val enabled: Boolean.
val maxRetries: Int,
val checkInterval: FiniteDuration,
val factor: Double)
```
Or you mean use connectionCheckerEnabled flag instead ConnectionCheckerSettings . And parse configs on actor start. But if config values invalid and ConnectionChecker actor fall with ActorCreationException then ConsumerActor and streams continue work like nothing happens

Reply:
Hmmm.. Maybe first case make sense because we provide default parameters and config must be valid. But I have feeling that someone would create actorSystem using ConfigFactory.parseString and even if they don't want use ConnectionChecker them must provide `connection-checker` configs

Reply:
Even the use of `ConfigFactory.parseString` needs to be resolved with `reference.conf`. New required settings can be introduced even in patch releases.

Reply:
@ennru Ok, then reimplement like
```
ConnectionCheckerSettings(
val enabled: Boolean.
val maxRetries: Int,
val checkInterval: FiniteDuration,
val factor: Double)
```
?

Reply:
Yes, please.